packages/headless/src/features/search/search-actions.ts

```diff
@@ -147,14 +157,51 @@ export const executeSearch = createAsyncThunk<
   }
 );
 
+export const fetchMoreResults = createAsyncThunk<
+  ExecuteSearchThunkReturn,
+  void,
+  AsyncThunkSearchOptions<StateNeededByExecuteSearch>
+>(
+  'search/fetchMoreResults',
+  async (
+    _,
+    {getState, dispatch, rejectWithValue, extra: {searchAPIClient}}
+  ) => {
+    const state = getState();
+    addEntryInActionsHistory(state);
+    const fetched = await fetchFromAPI(
+      searchAPIClient,
+      state,
+      buildFetchMoreRequest(state)
+    );
+
+    if (isErrorResponse(fetched.response)) {
+      return rejectWithValue(fetched.response.error);
+    }
+
+    dispatch(snapshot(extractHistory(state)));
```

Comment:
I don't think we should snapshot this as an entry in history, for the same reason as above.